lib/spack/spack/test/modules/lmod.py

```diff
@@ -59,10 +59,10 @@ def test_file_layout(self, compiler, provider, factory, module_configuration):
         # is transformed to r"Core" if the compiler is listed among core
         # compilers
         # Check that specs listed as core_specs are transformed to "Core"
-        if compiler == "clang@3.3" or spec_string == "mpich@3.0.1":
+        if compiler == "clang@=3.3" or spec_string == "mpich@3.0.1":
```

Comment:
could these just be `satisfies()`?  I'd almost rather force satisfies than make people remember that they need `@=` everywhere.

Reply:
`compiler` is a string here :p